### PR TITLE
Create survey on form submit

### PR DESCRIPTION
**Pull request by dipshit**
I did a few things here:

* Added tests for json ser and deser
* Format the `options` on the form to be an array of strings, instead of a string with commas in it
* changed toStrings to not sort, instead I sort or call `containsString` in the tests
* removed some unused java imports
* fixed a bug when comparing `Answer` classes
* added json annotations for setters and getters on the question subclasses

it works tho

Screen Shot 2020-03-25 at 4 51 28 PM
Screen Shot 2020-03-25 at 4 51 38 PM

---
### After submit:

Screen Shot 2020-03-25 at 4 53 10 PM

---

## Reviewing

You should probably use `&w=1` because I used a plugin called `Prettier` to format the js, and I reformatted code in the IDE.

I worked off of @bkzhang 's work with `jackson`, and I added json annotations based on the type annotations he added.

### CODE REVIEW

File: src/main/java/com/example/survey/controllers/SurveyController.java [hunk 1]
```diff
@@ -37,6 +37,9 @@ public Survey getSurvey(@RequestParam("id") long id) {
     @ResponseBody
     public Survey newSurvey(@RequestBody Survey survey) {
         this.surveyRepo.save(survey);
+        for (Question q : survey.questions) {
+            System.out.println("" + q.toString());
+        }
```
Comment (JackyChiu) [inline src/main/java/com/example/survey/controllers/SurveyController.java]: did you mean to remove this log?

File: src/test/java/com/example/survey/models/QuestionTest.java [hunk 2]
```diff
@@ -22,4 +29,49 @@ void testEquals() {
...
     }
+
+    @Test
+    void testTextQuestionJsonSerialization() {
+        question.id = 3;
+        question.addAnswer(new Answer("Joe"));
+        question.addAnswer(new Answer("Jim"));
+        try {
+            String result = new ObjectMapper().writeValueAsString(question);
+            assertThat(result, containsString("\"question\":\"What is your name?\""));
```
Comment (JackyChiu) [inline src/test/java/com/example/survey/models/QuestionTest.java]: I think you should be able to use `jsonPath` here instead, similar to the survey controller tests

Comment (JackyChiu) [reply src/test/java/com/example/survey/models/QuestionTest.java]: ^ applies to all tests that have an expected json value

Comment (dipshit) [reply src/test/java/com/example/survey/models/QuestionTest.java]: The same package can't be used. I tried importing [JsonPath](https://github.com/json-path/JsonPath) instead, but I could not get that working here.

**Pull request merged**